crates/sui-json-rpc/src/read_api.rs

```diff
@@ -143,6 +148,25 @@ impl ReadApiServer for ReadApi {
             .cloned()
             .map_or(cursor, |o_info| Some(o_info.object_id));
 
+        // filter here by filter options until we index tables by those options
+        match filter {
+            Some(SuiObjectDataFilter::StructType(struct_tag)) => {
+                objects.retain(|obj| obj.type_.to_string() == struct_tag.to_string())
+            }
+            Some(SuiObjectDataFilter::MoveModule { package, module }) => objects.retain(|obj| {
+                if let ObjectType::Struct(MoveObjectType::Other(ref tag)) = obj.type_ {
+                    tag.address.to_string() == package.to_string()
+                        && tag.module.to_string() == module.to_string()
```

Comment:
```suggestion
                if let ObjectType::Struct(o)) = obj.type_ {
                    tag.address() == package
                        && tag.module() == module
```

Reply:
Let's just keep the string checks actually because converting object types involve writing a bunch of new helpers.